app/models/course.rb

```diff
@@ -366,7 +366,7 @@ def self.get_by_codes(year, provider_code, course_code)
   end
 
   def generate_name
-    services[:generate_course_name].execute(course: self)
+    Courses::GenerateCourseNameService.call(course: self)
   end
```

Comment:
Maybe we can drop this `generate_name` all together

```
course.name = Courses::GenerateCourseNameService.call(course:)

```

Reply:
It's used in a few other places. I think the api it provides (via the course model) is probably easier to call than `Courses::GenerateCourseNameService.call(course:) at the moment